ui/app/components/http-requests-bar-chart.js

```diff
@@ -138,9 +153,34 @@ export default Component.extend({
       .merge(barsEnter)
       .attr('width', xScale.bandwidth())
       .attr('height', counter => height - yScale(counter.total))
-      // the offset between each bar
       .attr('x', counter => xScale(counter.start_time))
       .attr('y', counter => yScale(counter.total));
+
+    bars.exit().remove();
+
+    // render transparent bars and bind the tooltip to them
+    // since we cannot bind the tooltip to the actual bars
+    const shadowBarsContainer = d3.select('.shadow-bars');
+
+    const shadowBars = shadowBarsContainer.selectAll('.bar').data(parsedCounters, c => +c.start_time);
+
+    const shadowBarsEnter = shadowBars
+      .enter()
+      .append('rect')
+      .attr('class', 'bar')
+      .on('mouseenter', tip.show)
+      .on('mouseleave', tip.hide);
```

Comment:
I'm guessing all this is fine, but just incase these listener callbacks need rebinding so `this` is correct in `tip.show` etc. I don't have a lot of context on `d3` so I'm assuming this is fine, thought I'd mention just incase.